version: '3.8'

services:
  django:
    build: .
    volumes:
      - .:/opt/services/django/src
    env_file:
      - .env
    networks:
      - nginx_network
      - postgres_network
    depends_on:
      - postgres
  
  nginx:
    image: nginx:1.18.0
    volumes:
      - ./config/nginx/conf.d:/etc/nginx/conf.d
    depends_on:
      - django
    networks:
      - nginx_network
    ports:
      - 8000:80

  postgres:
    image: postgres:12.3
    volumes:
      - postgres_volume:/var/lib/postgresql/data
    env_file:
      - config/postgres/.env
    networks:
      - postgres_network
    ports:
      - '5432:5432'
    
  datadog:
    links:
      - django
      - nginx
      - postgres
    image: datadog/agent:7.19.2
    env_file:
      - config/datadog/.env
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /proc/:/host/proc/:ro
      - /sys/fs/cgroup:/host/sys/fs/cgroup:ro

networks:
  nginx_network:
    driver: bridge
  postgres_network:
    driver: bridge

volumes:
  postgres_volume: